File: ambeth/Ambeth.IoC/ambeth/accessor/AbstractAccessor.cs
using De.Osthus.Ambeth.Typeinfo;
using System;

namespace De.Osthus.Ambeth.Accessor
{
    public abstract class AbstractAccessor
    {
        protected AbstractAccessor(Type type, IPropertyInfo property)
	    {
		    // Intended blank
	    }

        public abstract bool CanRead { get; }

        public abstract bool CanWrite { get; }

        public abstract Object GetValue(Object obj, bool allowNullEquivalentValue);

	    public abstract Object GetValue(Object obj);

	    public abstract void SetValue(Object obj, Object value);

        public int GetIntValue(Object obj)
        {
            return ((int)GetValue(obj, true));
        }

        public void SetIntValue(Object obj, int value)
        {
            SetValue(obj, value);
        }
    }
}

File: jambeth/jambeth-ioc/src/main/java/de/osthus/ambeth/accessor/AbstractAccessor.java
package de.osthus.ambeth.accessor;

import de.osthus.ambeth.typeinfo.IPropertyInfo;

public abstract class AbstractAccessor
{
	protected AbstractAccessor(Class<?> type, IPropertyInfo property)
	{
		// Intended blank
	}

	public abstract boolean canRead();

	public abstract boolean canWrite();

	public abstract Object getValue(Object obj, boolean allowNullEquivalentValue);

	public abstract Object getValue(Object obj);

	public abstract void setValue(Object obj, Object value);

	public int getIntValue(Object obj)
	{
		return ((Number) getValue(obj, true)).intValue();
	}

	public void setIntValue(Object obj, int value)
	{
		setValue(obj, Integer.valueOf(value));
	}
}
